packages/wdio-allure-reporter/src/reporter.ts

```diff
@@ -1,82 +1,313 @@
+
+        if (error) {
+            currentSpec.detailsMessage = error.message
+            currentSpec.detailsTrace = error.stack
+        }
+
+        if (currentSpec instanceof AllureTest) {
+            currentSpec.endTest()
+        } else {
+            currentSpec.endStep()
+        }
+    }
+
+    _startStep(testTitle: string) {
+        if (!this.currentAllureSpec) {
+            throw new Error("There isn't any active test!")
+        }
+
+        const newStep = this.currentAllureSpec!.startStep(testTitle)
+
+        this._runningUnits.push(newStep)
+    }
+
+    setCaseParameters(cid: string | undefined) {
+        if (!this.currentTest) {
+            return
+        }
+
+        if (!this._isMultiremote) {
+            const caps = this._capabilities as Capabilities.DesiredCapabilities
+            const { browserName, deviceName, desired, device } = caps
+            let targetName = device || browserName || deviceName || cid
+
+            // custom mobile grids can have device information in a `desired` cap
+            if (desired && desired.deviceName && desired.platformVersion) {
+                targetName = `${device || desired.deviceName} ${desired.platformVersion}`
+            }
+
+            const browserstackVersion = caps.os_version || caps.osVersion
+            const version = browserstackVersion || caps.browserVersion || caps.version || caps.platformVersion || ''
+            const paramName = (deviceName || device) ? 'device' : 'browser'
+            const paramValue = version ? `${targetName}-${version}` : targetName
+
+            if (!paramValue) {
+                return
+            }
+
+            this.currentTest.addParameter(paramName, paramValue)
+        } else {
+            this.currentTest.addParameter('isMultiremote', 'true')
+        }
+
+        // Allure analytics labels. See https://github.com/allure-framework/allure2/blob/master/Analytics.md
+        this.currentTest.addLabel(LabelName.LANGUAGE, 'javascript')
+        this.currentTest.addLabel(LabelName.FRAMEWORK, 'wdio')
+
+        if (cid) {
+            this.currentTest.addLabel(LabelName.THREAD, cid)
+        }
+
+        if (!this.currentSuite) {
+            return
+        }
+
+        // TODO: need to add ability to get labels from allure entitites
+        // @ts-ignore
+        const isFeaturePresent = this.currentTest.wrappedItem.labels.some((label: Label) => label.name === LabelName.FEATURE)
+
+        if (isFeaturePresent) {
+            return
+        }
+
+        this.currentTest.addLabel(LabelName.FEATURE, this.currentSuite.name)
+    }
+
+    isScreenshotCommand(command: CommandArgs) {
+        const isScrenshotEndpoint = /\/session\/[^/]*(\/element\/[^/]*)?\/screenshot/
+
+        return (
+            // WebDriver protocol
+            (command.endpoint && isScrenshotEndpoint.test(command.endpoint)) ||
+            // DevTools protocol
+            command.command === 'takeScreenshot'
+        )
+    }
+
+    getLabels({
```

Comment:
The function doesn't use this. It will be fine to move it from the reporter class to the utils and cover it with unit tests if needed..

Reply:
:+1: